### PR TITLE
Configuration to produce ME0 pseudo stubs for Phase-2

### PR DESCRIPTION
**Pull request by dildick**
In PRs https://github.com/cms-sw/cmssw/pull/18229, https://github.com/cms-sw/cmssw/pull/18660 and https://github.com/cms-sw/cmssw/pull/19254 I have been putting various objects and producers in place to build a realistic L1 ME0 trigger stub producer. It could take a while before we have something that works, since the `ME0Motherboard` will likely be as complicated as the `CSCCathodeLCTProcessor` class.

In the mean time we have been running L1 simulations for the muon upgrade TDR on prompt muon and displaced muon triggers with so-called ME0 pseudo stubs. These are not actual `ME0TriggerDigi` objects, but rather `ME0Segments` produced from pseudo digis with a 2x coarser resolution than what is used in the offline. This is a reasonable emulation a future ME0 trigger. In this PR I put in place configuration fragments in order to produce such pseudo stubs. It is placed in the `ME0Trigger` package for now. 

@kpedro88 @tahuang1991 @nickmccoll

### CODE REVIEW

**Comment by kpedro88** [comment]
please test

File: L1Trigger/ME0Trigger/python/me0TriggerPseudoDigis_cff.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+## pseudo pads are created from pseudo digis with 192 strips instead of 384
+## the rechits are a necessary intermediate step before the pseudo pads are used
+## as input to build pseudo stubs
+
+from SimMuon.GEMDigitizer.muonME0PseudoReDigis_cfi import *
+from RecoLocalMuon.GEMRecHit.me0RecHits_cfi import *
+from RecoLocalMuon.GEMSegment.me0Segments_cfi import *
+
+simMuonME0PseudoReDigis192 = simMuonME0PseudoReDigis.clone(
+    numberOfStrips = cms.uint32(192)
```
Comment (nickmccoll) [inline L1Trigger/ME0Trigger/python/me0TriggerPseudoDigis_cff.py]: This only has an effect in a single partition geometry. 
See: https://github.com/cms-sw/cmssw/blob/35ad0e0d8f48fadc1782b97e3276f1c36f5feafe/SimMuon/GEMDigitizer/python/muonME0PseudoReDigis_cfi.py#L6

I have code to make different strip multiplicities work in multi-partition geometries, and can share it if it is wanted

File: L1Trigger/ME0Trigger/python/me0TriggerPseudoDigis_cff.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+)
+me0RecHits192 = me0RecHits.clone(
+    me0DigiLabel = cms.InputTag("simMuonME0PseudoReDigis192")
+)
+me0TriggerPseudoDigis = me0Segments.clone(
+    me0RecHitLabel = cms.InputTag("me0RecHits192")
```
Comment (nickmccoll) [inline L1Trigger/ME0Trigger/python/me0TriggerPseudoDigis_cff.py]: If you change the strip granularity, you have to change the strip matching window in the segment reconstruction...see:
https://github.com/cms-sw/cmssw/blob/35ad0e0d8f48fadc1782b97e3276f1c36f5feafe/SimMuon/GEMDigitizer/python/muonME0PseudoReDigis_cfi.py#L6
https://github.com/cms-sw/cmssw/blob/35ad0e0d8f48fadc1782b97e3276f1c36f5feafe/RecoLocalMuon/GEMSegment/python/ME0SegmentsRU_cfi.py#L10

Comment (dildick) [inline L1Trigger/ME0Trigger/python/me0TriggerPseudoDigis_cff.py]: right I forgot about that line...

**Comment by dildick** [comment]
Step 3 seems to fail because of

<PRE>
  File "/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_2_X_2017-06-21-1100/python/Configuration/StandardSequences/Reconstruction_cff.py", line 35, in <module>
    from RecoLocalTracker.Configuration.RecoLocalTracker_cff import *
  File "/cvmfs/cms-ib.cern.ch/nweek-02477/slc6_amd64_gcc530/cms/cmssw/CMSSW_9_2_X_2017-06-21-1100/python/RecoLocalTracker/Configuration/RecoLocalTracker_cff.py", line 21, in <module>
    from RecoLocalTracker.Phase2ITPixelClusterizer.Phase2ITPixelClusterizer_cfi import *
ImportError: No module named Phase2ITPixelClusterizer.Phase2ITPixelClusterizer_cfi
</PRE>

File: L1Trigger/ME0Trigger/python/me0TriggerPseudoDigis_cff.py (hunk 1 above)
Comment (dildick) [inline L1Trigger/ME0Trigger/python/me0TriggerPseudoDigis_cff.py]: @nickmccoll could you make a pull request with the relevant updates to this branch?

**Comment by kpedro88** [comment]
@dildick the IB is broken, we'll wait until a fixed build comes out

File: L1Trigger/ME0Trigger/python/me0TriggerPseudoDigis_cff.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+## configuration to build fast L1 ME0 trigger stubs
+## pseudo pads are created from pseudo digis with 192 strips instead of 384
+## the rechits are a necessary intermediate step before the pseudo pads are used
+## as input to build pseudo stubs
+
+from SimMuon.GEMDigitizer.muonME0PseudoReDigis_cfi import *
+from RecoLocalMuon.GEMRecHit.me0RecHits_cfi import *
+from RecoLocalMuon.GEMSegment.me0Segments_cfi import *
+
+simMuonME0PseudoReDigis192 = simMuonME0PseudoReDigis.clone(
```
Comment (kpedro88) [inline L1Trigger/ME0Trigger/python/me0TriggerPseudoDigis_cff.py]: The "192" naming scheme seems very contingent to me. I would call it something more general like "Coarse" or "Half" that indicates it is less granular without using a specific number.

**Comment by davidlange6** [comment]
please test

**Comment by davidlange6** [comment]
please test

**Comment by davidlange6** [comment]
please test

**Comment by smuzaffar** [comment]
please test

**Comment by kpedro88** [comment]
```
  File "/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_2_X_2017-06-22-1000/python/L1Trigger/ME0Trigger/me0TriggerPseudoDigis_cff.py", line 25, in <module>
    nStrips = simMuonME0PseudoReDigisCoarse.numberOfStrips/2
TypeError: unsupported operand type(s) for /: 'uint32' and 'int'
```

**Comment by dildick** [comment]
shoot. can't divide by two?

**Comment by kpedro88** [comment]
I think you can do this:
```
nStrips = cms.uint32(simMuonME0PseudoReDigisCoarse.numberOfStrips.value()/2)
```

**Comment by kpedro88** [comment]
Or you can omit the surrounding `cms.uint32()` since `nStrips` is an intermediate value

**Comment by kpedro88** [comment]
please test

**Comment by kpedro88** [comment]
+1
no changes in monitored quantities

**Comment by civanch** [comment]
+1

**Comment by Dr15Jones** [comment]
+1
just random number

**Comment by rekovic** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**